shared/trackingHooks.tsx

```diff
@@ -13,32 +13,39 @@ export const abExperiments = {
 /**
  * Fetch and return the user's anonymous ID.
  */
-export const useAnonymousID = (): { anonymousID: string; existing: boolean } => {
+export const useAnonymousID = (): {
+  anonymousID: string;
+  existing: boolean;
+} => {
+  // prevent race condition on re-renders
+  const cached = useRef<any>(false);
   const [anonymousID, setAnonymousID] = useCookie("inngest_anonymous_id");
 
-  // TODO: remove this once sufficient time has passed for users to get the new cookie.
-  // If the user has a legacy anonymous ID, migrate it to the new cookie.
-  const [legacyAnonymousID, _, deleteLegacyAnonymousID] = useLocalStorage<string>("inngest-anon-id");
-  if (!anonymousID && legacyAnonymousID) {
-    const sixMonthsFromNow = new Date(Date.now() + 6 * 30 * 24 * 60 * 60 * 1000);
-    setAnonymousID(legacyAnonymousID, { domain: process.env.NEXT_PUBLIC_HOSTNAME, path: "/", expires: sixMonthsFromNow, sameSite: 'lax' });
-    deleteLegacyAnonymousID();
-  }
+  if (!cached.current) {
+    if (!anonymousID) {
+      const newAnonymousID = uuid();
+      const sixMonthsFromNow = new Date(
+        Date.now() + 6 * 30 * 24 * 60 * 60 * 1000
+      );
+      setAnonymousID(newAnonymousID, {
+        domain: process.env.NEXT_PUBLIC_HOSTNAME,
+        path: "/",
+        expires: sixMonthsFromNow,
+        sameSite: "lax",
+      });
 
-  if (!anonymousID) {
-    const newAnonymousID = uuid();
-    const sixMonthsFromNow = new Date(Date.now() + 6 * 30 * 24 * 60 * 60 * 1000);
-    setAnonymousID(newAnonymousID, { domain: process.env.NEXT_PUBLIC_HOSTNAME, path: "/", expires: sixMonthsFromNow, sameSite: 'lax' });
-    return {
-      anonymousID: newAnonymousID,
-      existing: false,
-    };
+      cached.current = {
+        anonymousID: newAnonymousID,
+        existing: false,
+      };
```

Comment:
Does this just help when this hook is called multiple times on a single page render so it doesn't perform this multiple times and overwrite the value? 

How does this work with page loads/transitions? does existing change to true?

Reply:
> Does this just help when this hook is called multiple times on a single page render so it doesn't perform this multiple times and overwrite the value?

Exactly, this avoids the race condition of the hook being called multiple times when rendering the page at the initial load.
Without the use of this `cached` ref, the hook is called multiple times in a row, ending in never flagging a user as new.


> How does this work with page loads/transitions? does existing change to true?
The client-side routing changes are handled here, I believe: https://github.com/inngest/website/blob/cd6bdf76519623f6de9c75e3656e8eea8448e76e/pages/_app.tsx#L96